### PR TITLE
Command explanation update

### PR DESCRIPTION
**Pull request by kbristow**
This updates the command reference section of the docs splitting it into two sections:

1. Quick Reference - This groups the commands by functionality and provides a small description of the what the command does, similarly to `sub help`.
2. Detailed Reference - This section has been added and thorough explanations of how each command works have been included.

This aims to provide additional clarity for teams using Subatomic to avoid the blackbox and 'it does magic' feel.

### CODE REVIEW

File: docs/quantum-mechanic/command-reference.md [hunk 1]
```diff
@@ -1,76 +1,352 @@
...
+
+---
+#### `sub remove team member` 
+This command is used to remove a member from a Team. When removing a member from a team, the following actions are performed:
+
+1. Remove the Subatomic association between the user and the Team.
+2. Remove the user's permissions from all OpenShift namespaces associated to the Team.
+3. Remove the user's permissions from the Bitbucket project's associated to the Team.
+4. Kick the user from the Team's Slack channel.
+
```
Comment (KieranHons) [inline docs/quantum-mechanic/command-reference.md]: Perhaps a note about removing team owners?

**Note**: Team Owners can only be removed by a Team Owner (including themself) and may not remove themself if there are no other Team Owners in the team.

File: docs/quantum-mechanic/command-reference.md [hunk 2]
```diff
@@ -1,76 +1,352 @@
...
+2. Generate a Jenkinsfile that will be used to deploy the application into prod. The Jenkinsfile is the added to your Application repository if it does not exist. The name of the Jenkinsfile is `Jenkinsfile.[${pipelineTag}.]${environmentPostfix}` where `[${pipelineTag}.]` is not present if the deployment's owning pipeline has no tag.
+3. Create a Jenkins build job that uses the generated Jenkinsfile to deploy your Application. The build job is also added to the Jenkins Job View which holds all the Jenkins jobs related the Application being deployed.
+
+---
+#### `sub request devops environment` 
+A DevOps environment is an OpenShift namespace which acts as shared environment for common infrastructure across all of a Team's Projects. This by default includes a Jenkins instance used to build all Applications and Libraries in a Team's associated Projects and includes all the BuildConfigs used to run the S2I builds for the Team. Additionally infrastructure such as a Spring Cloud Config server or Key Cloak server would be added to the DevOps environment and shared across all Project environments. When running this command, the following actions will be performed:
```
Comment (KieranHons) [inline docs/quantum-mechanic/command-reference.md]:
```suggestion
A DevOps environment is an OpenShift namespace which acts as a shared environment for common infrastructure across all of a Team's Projects. This by default includes a Jenkins instance used to build all Applications and Libraries in a Team's associated Projects and includes all the BuildConfigs used to run the S2I builds for the Team. Additional infrastructure such as a Spring Cloud Config server or Key Cloak server would be added to the DevOps environment and shared across all Project environments. When running this command, the following actions will be performed: 
```

File: docs/quantum-mechanic/command-reference.md [hunk 3]
```diff
@@ -1,76 +1,352 @@
...
+
+---
+#### `sub link team channel`
+This command allows a user to link their Team to a new Slack channel for notification purposes. The Slack channel needs to exist for this command to work correctly. This is particularly useful if the currently associated Team channel was renamed, if you want to use a private slack channel(\*) for the team notifications, or if you have an existing channel you wish to use for the notifications and Subatomic control flow interaction. When linking a Team to a new Slack channel the following actions are performed:
+1. Invite the Members of the Team to the Slack channel.
+2. Invite the Atomist Bot to the Team Channel.
+
+The Atomist Bot should present a generic message about creating a DevOps environment whenever it joins a new channel that is associated to a Team.
+
+**\*Note:** If you want to use a private Slack channel, the Atomist bot cannot be invited automatically. You will need to manually invite the bot the to channel using `\invite @Atomist` once you have linked the Team to the Slack channel.
```
Comment (KieranHons) [inline docs/quantum-mechanic/command-reference.md]: Why does this Note have \*

File: docs/quantum-mechanic/command-reference.md [hunk 4]
```diff
@@ -1,76 +1,352 @@
...
+
+---
+#### `sub request project environments` 
+After creating a Subatomic Project, a user can request that the Project have OpenShift environments provisioned for it's use. When requesting these environments, if the Project has not had it's build/dev and release pipelines defined, the Project will be assigned a default pipeline structure. The following actions are then performed:
```
Comment (KieranHons) [inline docs/quantum-mechanic/command-reference.md]:
```suggestion
After creating a Subatomic Project, a user can request that the Project have OpenShift environments provisioned for its use. When requesting these environments, if the Project has not had its build/dev and release pipelines defined, the Project will be assigned a default pipeline structure. The following actions are then performed:
```

File: docs/quantum-mechanic/command-reference.md [hunk 5]
```diff
@@ -1,76 +1,352 @@
...
+    3. Create a Pod Network between each OpenShift Project Namespace and the Team's DevOps namespace. This allows the Applications in the Project namespaces to access Application's in the DevOps space via service URI's (e.g. accessing the DevOps Config Server).
+2. Configure the Project in Jenkins:
+    1. Create the build folder for the Project. The build folder uses the Project's name to name the build folder.
+    2. Create the environment name credentials within the build folder/job domain. These credentials store the OpenShift namespaces for each environment in your pipeline. E.g. "DEV_PROJECT" - Dev OpenShift Project, "TZA_UAT_PROJECT" - UAT OpenShift project for the TZA deployment pipeline.
+    3. Create the Bitbucket authorization credential. This is used by Jenkins to clone sources from the configured Bitbucket instance.
+
+---
+#### `sub request project prod` 
+In order to promote Applications' to production, you need to have production environments available to deploy the Application's into. This command will allow users to provision these production environments for a selected Project and deployment pipeline. The process follows the following approach:
```
Comment (KieranHons) [inline docs/quantum-mechanic/command-reference.md]:
```suggestion
In order to promote Applications to production, you need to have production environments available to deploy the Application's into. This command will allow users to provision these production environments for a selected Project and deployment pipeline. The process follows the following approach:
```

File: docs/quantum-mechanic/command-reference.md (hunk 3 above)
Comment (kbristow) [reply docs/quantum-mechanic/command-reference.md]: It is an exception to the normal flow as referenced by "if you want to use a private slack channel(\*)"

**Pull request merged**